src/quickpicks/schemas.ts

```diff
@@ -1,57 +1,84 @@
-import { getStorageManager } from "../storage";
-import { environmentQuickPick } from "./environments";
-
-export async function schemaQuickPick(): Promise<Schema | undefined> {
-  const cloudEnvironment = await environmentQuickPick();
-  if (!cloudEnvironment) {
-    return;
+import { Schema, SchemaType } from "../models/schema";
+import { CCloudResourcePreloader } from "../storage/ccloudPreloader";
+import { getResourceManager } from "../storage/resourceManager";
+
+/** Quickpick returning a string for what to use as a schema subject out of the preexisting options.
+ * @returns nonempty string if user chose an existing subject name.
+ * @returns empty string if user gestures to create a new subject.
+ * @returns undefined if user cancelled the quickpick.
+ *
+ */
+export async function schemaSubjectQuickPick(
+  schemaRegistryId: string,
+  onlyType: SchemaType | undefined = undefined,
+): Promise<string | undefined> {
+  // ensure that the resources are loaded before trying to access them
+  const preloader = CCloudResourcePreloader.getInstance();
+  await preloader.ensureCoarseResourcesLoaded();
+  await preloader.ensureSchemasLoaded(schemaRegistryId);
+
+  const schemas = await getResourceManager().getSchemasForRegistry(schemaRegistryId);
+
+  let schemaSubjects: string[] | undefined;
+
+  const latestVersionSchemaBySubject = new Map<string, Schema>();
+  if (schemas) {
+    // Crunch down to map of subject -> latest version'd Schema for said subject
+    for (const schema of schemas) {
+      // Skip if the caller asked for only a specific type of schema and this one doesn't match.
+      if (onlyType && schema.type !== onlyType) {
+        continue;
+      }
+      const latestVersionSchema = latestVersionSchemaBySubject.get(schema.subject);
+      if (!latestVersionSchema || schema.version > latestVersionSchema.version) {
+        latestVersionSchemaBySubject.set(schema.subject, schema);
+      }
+    }
+
+    schemaSubjects = Array.from(latestVersionSchemaBySubject.keys());
+    schemaSubjects.sort();
   }
 
-  const schemas: Schema[] | undefined = await getStorageManager().getWorkspaceState(
-    `${cloudEnvironment.id}.schemas`,
-  );
-  if (!schemas) {
-    vscode.window.showWarningMessage(`No schemas available for "${cloudEnvironment.name}".`);
-    return;
+  if (schemas === undefined) {
+    schemaSubjects = [];
   }
 
-  let schemaItems: vscode.QuickPickItem[] = [];
-  const schemaMap: Map<string, Schema> = new Map();
-  // sort by subject (ascending) and version (descending)
-  schemas.sort((a, b) => {
-    if (a.subject < b.subject) {
-      return -1;
-    } else if (a.subject > b.subject) {
-      return 1;
-    } else {
-      return b.version - a.version;
-    }
+  // Convert to quickpick items
+  let subjectItems: vscode.QuickPickItem[] = [];
+
+  // First entry to create a new schema / subject
+  const newSchemaLabel = "Create new schema / subject";
+  subjectItems.push({
+    label: newSchemaLabel,
   });
```

Comment:
What do you think about visually grouping these sections with separators? (Will probably need a short-term follow-up once local SR is ready to use a separator anyway.)
```suggestion
  // Convert to quickpick items, first entry to create a new schema / subject followed by a separator
  const newSchemaLabel = "Create new schema / subject";
  let subjectItems: vscode.QuickPickItem[] = [
    {
      label: newSchemaLabel,
      iconPath: new vscode.ThemeIcon("add"),
    },
    {
      kind: vscode.QuickPickItemKind.Separator,
      label: "Confluent Cloud",
    },
  ];
```
Example:
<img width="725" alt="image" src="https://github.com/user-attachments/assets/6875d5ed-c090-48d6-9056-fecf9eb33ac7">